.circleci/config.yml

```diff
@@ -63,20 +29,103 @@ jobs:
       - run:
           name: Build wrapper
           command: nix-shell --run "make -j 4 -C FV3 wrapper_build"
+      - save_cache:
+          name: Save nix build cache
+          key: nix-fv3gfs-fortran-{{ .Environment.CIRCLE_SHA1 }}
+          paths:
+            - FV3/
+jobs:
+  nix-full:
+    docker:
+      - image: nixos/nix
+    environment:
+      FV3CONFIG_CACHE_DIR: /tmp/.fv3config
+      GOOGLE_APPLICATION_CREDENTIALS: /tmp/key.json
+      FSSPEC_GS_REQUESTER_PAYS: vcm-ml
+    steps:
+      - nix
       - run:
-          name: Test
+          name: Test fortran
           command: nix-shell --run "pytest --native tests/pytest -v"
       - run:
           name: Test wrapper
-          command: nix-shell --run "make test_wrapper"
+          command: nix-shell --run "make -C FV3/wrapper test"
+  nix-basic:
+    docker:
+      - image: nixos/nix
+    environment:
+      FV3CONFIG_CACHE_DIR: /tmp/.fv3config
+      GOOGLE_APPLICATION_CREDENTIALS: /tmp/key.json
+      FSSPEC_GS_REQUESTER_PAYS: vcm-ml
+    steps:
+      - nix
+      - run:
+          name: Test fortran
+          command: nix-shell --run "pytest --native tests/pytest -v -m basic"
```

Comment:
Nit: Make similar rules in the root makefile as was done for the wrapper makefile to invoke particular marks.

Reply:
Indeed that could be convenient outside of CI as well.  I'll add some.